solidity/contracts/crypto/BigModExp.sol

```diff
@@ -0,0 +1,36 @@
+pragma solidity 0.4.24;
+
+/* Wrapper for bigModExp precompile contract */
+
+library BigModExp {
+  // Computes '(base**exponent) % modulus', via precompiled contract bigModExp.
+  // Returns (value, false) on success of bigModExp op, (_, true) on failure
+  //
+  // Cribbed from https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4
+  function wordBigModExp(uint256 _base, uint256 _exponent, uint256 _modulus)
+    public view returns (uint256, bool) {
+    uint256 callResult = 0;
+    uint256 callValue;
+    assembly {
+      // Lay out the arguments in memory, as the bigModExp contract expects.
+      // Since each argument is one word long, we increment the address by 0x20
+      // each time. First three arguments are the argument lengths, which are
+      // each 256 bits.
+      mstore(0x40, 0x20)  // Length of base argument
```

Comment:
Not sure you're supposed to just overwrite 0x40 like this, since it's the free space pointer?

Reply:
Thanks, you're right. That was one of the earlier experiments. It's better to create arrays in solidity code and use those as pointers in the assembly, like I do in `Curve.sol`.